addons/l10n_latam_check/models/l10n_latam_checkbook.py

```diff
@@ -0,0 +1,76 @@
+from odoo import fields, models, api, _
+import logging
+_logger = logging.getLogger(__name__)
+
+
+class L10nLatamCheckbook(models.Model):
+
+    _name = 'l10n_latam.checkbook'
+    _description = 'Checkbook'
+    _rec_name = 'sequence_id'
+    _order = 'active desc, range_to'
+
+    sequence_id = fields.Many2one(
+        comodel_name='ir.sequence',
+        string='Sequence',
+        domain=[('code', '=', 'own_check')],
+        copy=False,
+        help="Checks numbering sequence.",
+    )
+    next_number = fields.Integer(
+        related='sequence_id.number_next_actual', related_sudo=True,
+        readonly=False,
+    )
+    type = fields.Selection(
+        selection=[('deferred', 'Deferred'), ('currents', 'Currents'), ('electronic', 'Electronic')],
+        string='Check type',
+        default='deferred',
+        required=True,
+    )
```

Comment:
**Current** checkbook: checks that can be collected from the date that is created (no Check Payment Date needed)
**Deferred**: checks that can be collected on or after the Check Payment Date
**Electronic**: can be both, and it will depend on the check payment date > if check payment date = today() is current if check payment date > today() deferred
A tooltip based on my above comments can be added if needed (or improve it as well)

They are 3 different checkbooks and have their own sequences.

We need to have the check payment date for electronic and deferred but not for current.
Even though it was not used in the video, it is used by companies, usually, they use a mix of the 3 of them, as they are 3 different types of checkbooks, and you have even more than 1 deferred or current checkbook, e.g. 2 deferred checkbooks and 1 current checkbook, all active and available at the same time.

@william-andre @zaoral @jco-odoo @jjscarafia @fvz-odoo

Reply:
@zaoral add a helper with my comment